src/Allocator/Allocator.hs

```diff
@@ -0,0 +1,64 @@
+module Allocator.Allocator where
+
+import Env
+import Manager
+import Foundation
+import Control.Effect.Error
+import Control.Effect.State
+import Data.HashMap.Lazy
+import Control.Monad
+
+-- map of processes to its pages
+type PMap = HashMap ProcessId [Page 'Ram]
+type Ptr = Int
+newtype Size = Size { unSize :: CountOf Word8 } deriving (Eq, Show)
```

Comment:
`CountOf` is a type reserved for describing size from foundation. From logical point of view, you're creating `Size` type atop of size type.